cosmos/dataset.py

```diff
@@ -17,7 +17,7 @@ def get_dataset_alias_name(dag: DAG | None, task_group: TaskGroup | None, task_i
             dag_id = task_group.dag_id
         if task_group.group_id is not None:
             task_group_id = task_group.group_id
-            task_group_id = task_group_id.replace(".", "__")
+            task_group_id = task_group_id.split(".")[-1]
```

Comment:
Do we need to worry about any backward compatibility issues?

Reply:
We haven't released this module yet - we implemented it as part of the `DatasetAlias` that will be released in 1.7.0. Since TaskGroups are unique per DAG, I think it's an opportunity to simplify.

I also predicted that we could have a similar issue with task groups being resolved differently in initialization/execution - so this will hopefully avoid any potential issues in this area.